components/function-controller/internal/registry/repository.go

```diff
@@ -0,0 +1,79 @@
+package registry
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/distribution/registry/client/auth"
+	"github.com/docker/distribution/registry/client/transport"
+	dockertypes "github.com/docker/docker/api/types"
+	dockerregistry "github.com/docker/docker/registry"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+type RepositoryClient interface {
+	ListTags() ([]string, error)
+	GetImageTag(tag string) (*Tag, error)
+	DeleteImageTag(tagDigest digest.Digest) error
+}
+
+type repositoryClient struct {
+	ctx context.Context
+
+	namedImage reference.Named
+
+	tagSservice     distribution.TagService
```

Comment:
Is the double "S" (tag**Ss**ervice) in this name intentional?

Reply:
typo :)